# Validating Tissue Signatures {#validating-signatures}

The authors of *BioQC* have taken three independent approaches to show that their signatures are valid and biologically meaningful. However, they did not break down the predictive performance (*i.e.* is the signature able to identify its tissue) of each signature with quantitative performance measures. 

To address this, we independently derived signatures on the [GTEx](http://gtexportal.org) dataset using *gini-index*.  and performed both a 10-fold cross validation on the same dataset and a cross-species, cross-platform validation on a different dataset. To this end, we created the python package [pygenesig](https://github.com/grst/pygenesig), a framework to create and validate signatures. 

In this chapter, we

 * perform a 10-fold cross-validation on the GTEx dataset, calculating the sensitivity, specificity and precition for each signature. 
 * perform a cross-species, cross-platform validation of the signatures generated on the GTEx dataset
 * identify a set of tissues, that can be reliably and unabmigously identified with the *BioQC* method. 


## Data
 * The [**Genotype Tissue Expression (GTEx)**](http://gtexportal.org) project is a comprehensive resource of tissue-specific gene expression data. We use this dataset to derive tissue-specific signatures. The data is human only and was generated using Illumina sequencing. 
 * We use the [**GNF Mouse GeneAtlas V3**](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10246) as a control dataset to demonstrate that the gini-method is robust over multiple platforms and species. This dataset originates from mouse and was generated using the *Affymetrix Mouse Genome 430 2.0 Array (GPL1261)*.
    
## Cross-Validation of signatures on the GTEx dataset 

```{r child="_notebooks/validate_gini.md"}
```

## Cross-species Cross-platform validation on mouse
```{r child="_notebooks/validate_mouse.md"}
```
