## Sample Processing with BioQC {#sample-processing}

```{r setup, include=FALSE}
source("lib/knitr.R")
library(BioQC)
library(VennDiagram)
library(data.table)
library(grid)
library(gridBase)
library(stringr)
library(readr)
library(testthat)
source("lib/plots.R")
source("lib/db.R")
```

The following processes are ressource intensive, therefore we execute them on a high performance cluster (HPC). We use [chunksub](https://github.com/grst/chunksub) to distribute the [list of sample ids](#sample-list) to the workers. This involves four major steps which are also documented in the project's [Makefile](https://github.com/grst/BioQC_GEO_analysis/blob/master/Makefile). 

1. We download the studies with [GEOquery](https://bioconductor.org/packages/release/bioc/html/GEOquery.html) and store them as R [ExpressionSet](https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf) using the R script [geo_to_eset.R](https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R). For some series, the download is not successful. 
2. We annotated human orthologous genes for all studies using [ribiosAnnotation](https://github.com/Accio/ribios) in [annotate_eset.R](https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/annotate_eset.R). This is necessary as the tissue signatures are built on a human dataset. The annotation failes for species which are not in the *ribios* database. 
3. We run *BioQC* on these studies use [run_bioqc.R](https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/run_bioqc.R). 
4. Finally, we prefilter BioQC's results for having a p-value < 0.05 and [import them into the database](#import-bioqc-results). 


## Sample Post-selection

The failures during download and annotation reduce the number of samples available to our study.

<!-- since there are muliple gpl and gsm in a gse, there might've been some gsm 
inserted, that don't have a tissue annotated, albeit the gse was selected. -->

```{r after_failures, cache=TRUE}

sql_select = "
select /*+ USE_HASH(bs, bgg) parallel(16) */ count(distinct bg.gsm) as GSM
                                           , count(distinct bgg.gse) as GSE"
sql_from = "
from bioqc_bioqc_success bs
join bioqc_gsm bg
  on bg.gsm = bs.gsm
join bioqc_gse_gsm bgg
  on bgg.gsm = bs.gsm 
"
sql_where = ""
res = dbGetQuery(mydb, str_c(sql_select, sql_from, sql_where, sep="\n"))
kable(res)
```


### Excluding multi-channel microarrays

Multi channel microarrays date back to the early age of gene expression studies. They don't provide absolute gene expression data and are not meaningful outside their experimental context. We therefore exclude these experiments:
```{r exclude_multichannel, cache=TRUE}
sql_select2 = sql_select
sql_from2 = sql_from
sql_where2 = str_c(sql_where, "where channel_count = 1", sep="\n")
res = dbGetQuery(mydb, str_c(sql_select2, sql_from2, sql_where2, sep="\n"))
kable(res)
```

### Exclude non-mapped tissues
We exclude samples that have a tissue annotated, but it is not mapped to a [normalized tissue](#tissue-normalization). 

```{r normalize_tissue, cache=TRUE}
sql_select3 = sql_select2
sql_from3 = str_c(sql_from2, "
join bioqc_normalize_tissues bnt
  on bnt.tissue_orig = lower(bg.tissue_orig)", sep="\n")
sql_where3 = sql_where2
res = dbGetQuery(mydb, str_c(sql_select3, sql_from3, sql_where3, sep="\n"))
kable(res)
```

### Select organisms
We were interested in the organism distribution.
```{r filter_orga, cache=TRUE}
sql_select4 = str_c(sql_select3, ", bg.organism_ch1", sep="\n")
sql_from4 = sql_from3
sql_where4 = sql_where3
res = dbGetQuery(mydb, str_c(sql_select4, sql_from4, sql_where4, "
group by organism_ch1
order by gsm desc", sep="\n"))
kable(res)
```

Results suggest that it makes sense to limit the analysis to the three main organisms: *H. sapiens*, *M. musculus*, *R. norvegicus*. This makes also sense as these species are closesly related and therefore the signatures are more likely to translate within these species. 
We are left with the following amount of samples: 

```{r filter_orga2, cache=TRUE}
sql_select5 = sql_select3
sql_from5 = sql_from4
sql_where5 = str_c(sql_where4, "
  and organism_ch1 in ('Homo sapiens', 'Mus musculus', 'Rattus norvegicus')", sep="\n")
res = dbGetQuery(mydb, str_c(sql_select5, sql_from5, sql_where5, sep="\n"))
kable(res)
```

### Exclude normalized studies
```{r }
sql = "
select study_median
from bioqc_gse_gpl
where study_median is not null"

res = dbGetQuery(mydb, sql)
sm = res$STUDY_MEDIAN
hist(sm[sm > -5 & sm < 30], breaks=seq(-5, 30, .2))
```

We observe three peaks, the first and second
have been normalized in some way. We focus on the studies 
with a median between 3 and 9. 

*TODO do proper fit of distributions and rationale based on standard deviation* 

```{r filter_median, cache=TRUE}
sql_select6 = sql_select5
sql_from6 = str_c(sql_from5, "
  join bioqc_gse_gpl bgl on 
  bgg.gse = bgl.gse and bg.gpl = bgl.gpl", sep="\n")
sql_where6 = str_c(sql_where5, "
  and study_median between 3 and 9", sep="\n")
res = dbGetQuery(mydb, str_c(sql_select6, sql_from6, sql_where6, sep="\n"))
kable(res)
```

This is the 'background' of samples on which we test for tissue contamination. 
We store the selected samples alongside with meta information required for the analysis in the materialized view `BIOQC_SELECTED_SAMPLES` in our DBS. [Here](https://github.com/grst/BioQC_GEO_analysis/blob/master/db/views/sample_selection.sql) is the SQL script generating the materialized view. 


### Tissue abundance

The following table shows how many samples are available for each tissue: 
```{r tissues, cache=TRUE}
sqlTissue = "
select /*+ parallel(16) */ tissue, count(distinct gsm) as samples from bioqc_selected_samples
group by tissue
order by samples desc"
resTissue = dbGetQuery(mydb, sqlTissue)
kable(resTissue)
```
