<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title></title>
  <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">
  <meta name="description" content="">
  <meta name="generator" content="bookdown 0.3 and GitBook 2.6.7">

  <meta property="og:title" content="" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="setup-database.html">
<link rel="next" href="contamination-analysis.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>


  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> BioQC GEO Analysis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#the-experiment-in-brief"><i class="fa fa-check"></i><b>1.2</b> The Experiment in Brief</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="validating-signatures.html"><a href="validating-signatures.html"><i class="fa fa-check"></i><b>2</b> Validating Tissue Signatures</a><ul>
<li class="chapter" data-level="2.1" data-path="validating-signatures.html"><a href="validating-signatures.html#data"><i class="fa fa-check"></i><b>2.1</b> Data</a></li>
<li class="chapter" data-level="2.2" data-path="validating-signatures.html"><a href="validating-signatures.html#validation"><i class="fa fa-check"></i><b>2.2</b> Validation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="curating-data.html"><a href="curating-data.html"><i class="fa fa-check"></i><b>3</b> Curating Data</a><ul>
<li class="chapter" data-level="3.1" data-path="curating-data.html"><a href="curating-data.html#selecting-samples-by-metadata"><i class="fa fa-check"></i><b>3.1</b> Selecting Samples by Metadata</a></li>
<li class="chapter" data-level="3.2" data-path="curating-data.html"><a href="curating-data.html#normalize-tissues"><i class="fa fa-check"></i><b>3.2</b> Normalize Tissues</a></li>
<li class="chapter" data-level="3.3" data-path="curating-data.html"><a href="curating-data.html#tissue-signatures"><i class="fa fa-check"></i><b>3.3</b> Map tissues to signatures</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="setup-database.html"><a href="setup-database.html"><i class="fa fa-check"></i><b>4</b> Setup Database</a><ul>
<li class="chapter" data-level="4.1" data-path="setup-database.html"><a href="setup-database.html#tables-explained"><i class="fa fa-check"></i><b>4.1</b> Tables explained</a><ul>
<li class="chapter" data-level="4.1.1" data-path="setup-database.html"><a href="setup-database.html#geometadb"><i class="fa fa-check"></i><b>4.1.1</b> GEOmetadb</a></li>
<li class="chapter" data-level="4.1.2" data-path="setup-database.html"><a href="setup-database.html#bioqc"><i class="fa fa-check"></i><b>4.1.2</b> BioQC</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="setup-database.html"><a href="setup-database.html#import-geometadb"><i class="fa fa-check"></i><b>4.2</b> Import GEOmetadb</a><ul>
<li class="chapter" data-level="4.2.1" data-path="setup-database.html"><a href="setup-database.html#fix-foreign-key-constraints"><i class="fa fa-check"></i><b>4.2.1</b> Fix foreign key constraints</a></li>
<li class="chapter" data-level="4.2.2" data-path="setup-database.html"><a href="setup-database.html#extract-tissue-annotation"><i class="fa fa-check"></i><b>4.2.2</b> Extract Tissue annotation</a></li>
<li class="chapter" data-level="4.2.3" data-path="setup-database.html"><a href="setup-database.html#load-annotation-information"><i class="fa fa-check"></i><b>4.2.3</b> Load annotation information</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-data"><i class="fa fa-check"></i><b>4.3</b> Import BioQC data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="setup-database.html"><a href="setup-database.html#signatures"><i class="fa fa-check"></i><b>4.3.1</b> Signatures</a></li>
<li class="chapter" data-level="4.3.2" data-path="setup-database.html"><a href="setup-database.html#tissue-annotation"><i class="fa fa-check"></i><b>4.3.2</b> Tissue Annotation</a></li>
<li class="chapter" data-level="4.3.3" data-path="setup-database.html"><a href="setup-database.html#tissue-sets"><i class="fa fa-check"></i><b>4.3.3</b> Tissue Sets</a></li>
<li class="chapter" data-level="4.3.4" data-path="setup-database.html"><a href="setup-database.html#import-bioqc-results"><i class="fa fa-check"></i><b>4.3.4</b> BioQC results</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="setup-database.html"><a href="setup-database.html#store-results"><i class="fa fa-check"></i><b>4.4</b> Store results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sample-selection.html"><a href="sample-selection.html"><i class="fa fa-check"></i><b>5</b> Sample Selection and processing</a><ul>
<li class="chapter" data-level="5.1" data-path="sample-selection.html"><a href="sample-selection.html#sample-preselection"><i class="fa fa-check"></i><b>5.1</b> Sample Preselection</a><ul>
<li class="chapter" data-level="5.1.1" data-path="sample-selection.html"><a href="sample-selection.html#required-annotation"><i class="fa fa-check"></i><b>5.1.1</b> Required annotation</a></li>
<li class="chapter" data-level="5.1.2" data-path="sample-selection.html"><a href="sample-selection.html#sample-list"><i class="fa fa-check"></i><b>5.1.2</b> Export list of samples</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="sample-selection.html"><a href="sample-selection.html#sample-processing"><i class="fa fa-check"></i><b>5.2</b> Sample Processing with BioQC</a></li>
<li class="chapter" data-level="5.3" data-path="sample-selection.html"><a href="sample-selection.html#sample-post-selection"><i class="fa fa-check"></i><b>5.3</b> Sample Post-selection</a><ul>
<li class="chapter" data-level="5.3.1" data-path="sample-selection.html"><a href="sample-selection.html#excluding-multi-channel-microarrays"><i class="fa fa-check"></i><b>5.3.1</b> Excluding multi-channel microarrays</a></li>
<li class="chapter" data-level="5.3.2" data-path="sample-selection.html"><a href="sample-selection.html#exclude-non-mapped-tissues"><i class="fa fa-check"></i><b>5.3.2</b> Exclude non-mapped tissues</a></li>
<li class="chapter" data-level="5.3.3" data-path="sample-selection.html"><a href="sample-selection.html#select-organisms"><i class="fa fa-check"></i><b>5.3.3</b> Select organisms</a></li>
<li class="chapter" data-level="5.3.4" data-path="sample-selection.html"><a href="sample-selection.html#exclude-normalized-studies"><i class="fa fa-check"></i><b>5.3.4</b> Exclude normalized studies</a></li>
<li class="chapter" data-level="5.3.5" data-path="sample-selection.html"><a href="sample-selection.html#tissue-abundance"><i class="fa fa-check"></i><b>5.3.5</b> Tissue abundance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="contamination-analysis.html"><a href="contamination-analysis.html"><i class="fa fa-check"></i><b>6</b> Analyse Contamination</a><ul>
<li class="chapter" data-level="6.1" data-path="contamination-analysis.html"><a href="contamination-analysis.html#input-tables"><i class="fa fa-check"></i><b>6.1</b> Input Tables</a></li>
<li class="chapter" data-level="6.2" data-path="contamination-analysis.html"><a href="contamination-analysis.html#processing-steps-explained"><i class="fa fa-check"></i><b>6.2</b> Processing Steps explained</a><ul>
<li class="chapter" data-level="6.2.1" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_selected_samples_tset"><i class="fa fa-check"></i><b>6.2.1</b> BIOQC_SELECTED_SAMPLES_TSET</a></li>
<li class="chapter" data-level="6.2.2" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_res_tset"><i class="fa fa-check"></i><b>6.2.2</b> BIOQC_RES_TSET</a></li>
<li class="chapter" data-level="6.2.3" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_tissue_enrichment"><i class="fa fa-check"></i><b>6.2.3</b> BIOQC_TISSUE_ENRICHMENT</a></li>
<li class="chapter" data-level="6.2.4" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_tissue_enrichment2"><i class="fa fa-check"></i><b>6.2.4</b> BIOQC_TISSUE_ENRICHMENT2</a></li>
<li class="chapter" data-level="6.2.5" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_contam_stats"><i class="fa fa-check"></i><b>6.2.5</b> BIOQC_CONTAM_STATS</a></li>
<li class="chapter" data-level="6.2.6" data-path="contamination-analysis.html"><a href="contamination-analysis.html#bioqc_tissue_migration"><i class="fa fa-check"></i><b>6.2.6</b> BIOQC_TISSUE_MIGRATION</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="tissue-migration.html"><a href="tissue-migration.html"><i class="fa fa-check"></i><b>7</b> Tissue Migration</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sample-selection" class="section level1">
<h1><span class="header-section-number">5</span> Sample Selection and processing</h1>
<div id="sample-preselection" class="section level2">
<h2><span class="header-section-number">5.1</span> Sample Preselection</h2>
<p>Here, we document the sample selection process before running <em>BioQC</em>.</p>
<div id="required-annotation" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Required annotation</h3>
<p>A sample is <em>usable</em> for this study, if</p>
<ul>
<li>the gene symbos are annotated (requirement to run BioQC)</li>
<li>the tissue of origin is annotated (requirment to draw conclusions about contamination)</li>
</ul>
<p>We consider two approaches for annotating gene symbols:</p>
<ul>
<li>Using the Bioconductor <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html">AnnotationDbi</a> package. The GEOmetadb provides a mapping of the GPL identifier to these packages.</li>
<li>Using the <code>annotGPL=TRUE</code> option of <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOquery</a>’s <code>getGEO</code>. This requires an annotation file being available for the respective platform. We retrieved a list of the available annoataion files in an earlier step.</li>
</ul>
<p>We compare the two methods with respect to the amount of usable samples that we can get.</p>
<p>First we need to run the sql-script <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/db/views/annotation_stats.sql">annotation_stats</a> which will create views summarizing the amount of usable samples. From these views, we can calculate a Venn diagram:</p>
<p><img src="20_sample_preselection_files/figure-html/sample_filtering-1.png" width="768" style="display:block; margin: auto" style="display: block; margin: auto;" /></p>
<p>The <code>getGEO</code> method appears to be the more powerful method. Ideal would be a combination of the two, however, for the sake of simplicity, we stick to <code>getGEO</code>, loosing 499 studies (35602 samples).</p>
<p>This leaves us with the following filtering result:</p>
<table>
<thead>
<tr class="header">
<th align="left">comment</th>
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">total</td>
<td align="right">1945417</td>
<td align="right">73719</td>
</tr>
<tr class="even">
<td align="left">tissue annotated</td>
<td align="right">760798</td>
<td align="right">24267</td>
</tr>
<tr class="odd">
<td align="left">annotation file available</td>
<td align="right">768346</td>
<td align="right">31579</td>
</tr>
<tr class="even">
<td align="left">tissue and annotation file</td>
<td align="right">275206</td>
<td align="right">9632</td>
</tr>
</tbody>
</table>
</div>
<div id="sample-list" class="section level3">
<h3><span class="header-section-number">5.1.2</span> Export list of samples</h3>
<p>We store the respective gse identifiers in <code>results/gse_lists/gse_tissue_annotation.txt</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sqlGse =<span class="st"> &quot;</span>
<span class="st">select distinct gse</span>
<span class="st">from (</span>
<span class="st">  select *</span>
<span class="st">  from bioqc_studies_has_tissue</span>
<span class="st">  intersect</span>
<span class="st">    select *</span>
<span class="st">    from bioqc_studies_has_annot) u&quot;</span>
gse =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, sqlGse)
<span class="kw">writeLines</span>(gse$GSE, <span class="kw">file</span>(gse.file))</code></pre></div>

</div>
</div>
<div id="sample-processing" class="section level2">
<h2><span class="header-section-number">5.2</span> Sample Processing with BioQC</h2>
<p>The following processes are ressource intensive, therefore we execute them on a high performance cluster (HPC). We use <a href="https://github.com/grst/chunksub">chunksub</a> to distribute the <a href="sample-selection.html#sample-list">list of sample ids</a> to the workers. This involves four major steps which are also documented in the project’s <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/Makefile">Makefile</a>.</p>
<ol style="list-style-type: decimal">
<li>We download the studies with <a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html">GEOquery</a> and store them as R <a href="https://bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf">ExpressionSet</a> using the R script <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/geo_to_eset.R">geo_to_eset.R</a>. For some series, the download is not successful.</li>
<li>We annotated human orthologous genes for all studies using <a href="https://github.com/Accio/ribios">ribiosAnnotation</a> in <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/annotate_eset.R">annotate_eset.R</a>. This is necessary as the tissue signatures are built on a human dataset. The annotation failes for species which are not in the <em>ribios</em> database.</li>
<li>We run <em>BioQC</em> on these studies use <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/scripts/run_bioqc.R">run_bioqc.R</a>.</li>
<li>Finally, we prefilter BioQC’s results for having a p-value &lt; 0.05 and <a href="setup-database.html#import-bioqc-results">import them into the database</a>.</li>
</ol>
</div>
<div id="sample-post-selection" class="section level2">
<h2><span class="header-section-number">5.3</span> Sample Post-selection</h2>
<p>The failures during download and annotation reduce the number of samples available to our study.</p>
<!-- since there are muliple gpl and gsm in a gse, there might've been some gsm 
inserted, that don't have a tissue annotated, albeit the gse was selected. -->
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select =<span class="st"> &quot;</span>
<span class="st">select /*+ USE_HASH(bs, bgg) parallel(16) */ count(distinct bg.gsm) as GSM</span>
<span class="st">                                           , count(distinct bgg.gse) as GSE&quot;</span>
sql_from =<span class="st"> &quot;</span>
<span class="st">from bioqc_bioqc_success bs</span>
<span class="st">join bioqc_gsm bg</span>
<span class="st">  on bg.gsm = bs.gsm</span>
<span class="st">join bioqc_gse_gsm bgg</span>
<span class="st">  on bgg.gsm = bs.gsm </span>
<span class="st">&quot;</span>
sql_where =<span class="st"> &quot;&quot;</span>
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select, sql_from, sql_where, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">253210</td>
<td align="right">8083</td>
</tr>
</tbody>
</table>
<div id="excluding-multi-channel-microarrays" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Excluding multi-channel microarrays</h3>
<p>Multi channel microarrays date back to the early age of gene expression studies. They don’t provide absolute gene expression data and are not meaningful outside their experimental context. We therefore exclude these experiments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select2 =<span class="st"> </span>sql_select
sql_from2 =<span class="st"> </span>sql_from
sql_where2 =<span class="st"> </span><span class="kw">str_c</span>(sql_where, <span class="st">&quot;where channel_count = 1&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select2, sql_from2, sql_where2, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">235237</td>
<td align="right">7561</td>
</tr>
</tbody>
</table>
</div>
<div id="exclude-non-mapped-tissues" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Exclude non-mapped tissues</h3>
<p>We exclude samples that have a tissue annotated, but it is not mapped to a <a href="#tissue-normalization">normalized tissue</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select3 =<span class="st"> </span>sql_select2
sql_from3 =<span class="st"> </span><span class="kw">str_c</span>(sql_from2, <span class="st">&quot;</span>
<span class="st">join bioqc_normalize_tissues bnt</span>
<span class="st">  on bnt.tissue_orig = lower(bg.tissue_orig)&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
sql_where3 =<span class="st"> </span>sql_where2
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select3, sql_from3, sql_where3, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">135670</td>
<td align="right">3770</td>
</tr>
</tbody>
</table>
</div>
<div id="select-organisms" class="section level3">
<h3><span class="header-section-number">5.3.3</span> Select organisms</h3>
<p>We were interested in the organism distribution.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select4 =<span class="st"> </span><span class="kw">str_c</span>(sql_select3, <span class="st">&quot;, bg.organism_ch1&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
sql_from4 =<span class="st"> </span>sql_from3
sql_where4 =<span class="st"> </span>sql_where3
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select4, sql_from4, sql_where4, <span class="st">&quot;</span>
<span class="st">group by organism_ch1</span>
<span class="st">order by gsm desc&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
<th align="left">ORGANISM_CH1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">65769</td>
<td align="right">1201</td>
<td align="left">Homo sapiens</td>
</tr>
<tr class="even">
<td align="right">38096</td>
<td align="right">2267</td>
<td align="left">Mus musculus</td>
</tr>
<tr class="odd">
<td align="right">29909</td>
<td align="right">278</td>
<td align="left">Rattus norvegicus</td>
</tr>
<tr class="even">
<td align="right">1082</td>
<td align="right">24</td>
<td align="left">Macaca mulatta</td>
</tr>
<tr class="odd">
<td align="right">259</td>
<td align="right">7</td>
<td align="left">Macaca fascicularis</td>
</tr>
<tr class="even">
<td align="right">202</td>
<td align="right">2</td>
<td align="left">Mus musculus musculus x M. m. domesticus</td>
</tr>
<tr class="odd">
<td align="right">80</td>
<td align="right">2</td>
<td align="left">Cercocebus atys</td>
</tr>
<tr class="even">
<td align="right">57</td>
<td align="right">1</td>
<td align="left">Oryctolagus cuniculus</td>
</tr>
<tr class="odd">
<td align="right">36</td>
<td align="right">1</td>
<td align="left">Chlorocebus aethiops</td>
</tr>
<tr class="even">
<td align="right">32</td>
<td align="right">3</td>
<td align="left">Mus musculus domesticus</td>
</tr>
<tr class="odd">
<td align="right">25</td>
<td align="right">1</td>
<td align="left">Pan troglodytes</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="right">1</td>
<td align="left">Papio cynocephalus</td>
</tr>
<tr class="odd">
<td align="right">19</td>
<td align="right">1</td>
<td align="left">Mus spretus</td>
</tr>
<tr class="even">
<td align="right">18</td>
<td align="right">1</td>
<td align="left">Capra hircus</td>
</tr>
<tr class="odd">
<td align="right">16</td>
<td align="right">1</td>
<td align="left">Phodopus sungorus</td>
</tr>
<tr class="even">
<td align="right">12</td>
<td align="right">1</td>
<td align="left">Papio hamadryas</td>
</tr>
<tr class="odd">
<td align="right">12</td>
<td align="right">1</td>
<td align="left">Mus musculus musculus x M. m. castaneus</td>
</tr>
<tr class="even">
<td align="right">8</td>
<td align="right">1</td>
<td align="left">Macaca nemestrina</td>
</tr>
<tr class="odd">
<td align="right">6</td>
<td align="right">1</td>
<td align="left">Mus musculus musculus</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">1</td>
<td align="left">Mus musculus castaneus</td>
</tr>
<tr class="odd">
<td align="right">3</td>
<td align="right">1</td>
<td align="left">Mus sp.</td>
</tr>
</tbody>
</table>
<p>Results suggest that it makes sense to limit the analysis to the three main organisms: <em>H. sapiens</em>, <em>M. musculus</em>, <em>R. norvegicus</em>. This makes also sense as these species are closesly related and therefore the signatures are more likely to translate within these species. We are left with the following amount of samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select5 =<span class="st"> </span>sql_select3
sql_from5 =<span class="st"> </span>sql_from4
sql_where5 =<span class="st"> </span><span class="kw">str_c</span>(sql_where4, <span class="st">&quot;</span>
<span class="st">  and organism_ch1 in (&#39;Homo sapiens&#39;, &#39;Mus musculus&#39;, &#39;Rattus norvegicus&#39;)&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select5, sql_from5, sql_where5, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">133774</td>
<td align="right">3727</td>
</tr>
</tbody>
</table>
</div>
<div id="exclude-normalized-studies" class="section level3">
<h3><span class="header-section-number">5.3.4</span> Exclude normalized studies</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql =<span class="st"> &quot;</span>
<span class="st">select study_median</span>
<span class="st">from bioqc_gse_gpl</span>
<span class="st">where study_median is not null&quot;</span>

res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, sql)
sm =<span class="st"> </span>res$STUDY_MEDIAN
<span class="kw">hist</span>(sm[sm &gt;<span class="st"> </span>-<span class="dv">5</span> &amp;<span class="st"> </span>sm &lt;<span class="st"> </span><span class="dv">30</span>], <span class="dt">breaks=</span><span class="kw">seq</span>(-<span class="dv">5</span>, <span class="dv">30</span>, .<span class="dv">2</span>))</code></pre></div>
<p><img src="30_sample_postselection_files/figure-html/unnamed-chunk-1-1.png" width="672" style="display:block; margin: auto" style="display: block; margin: auto;" /></p>
<p>We observe three peaks, the first and second have been normalized in some way. We focus on the studies with a median between 3 and 9.</p>
<p><em>TODO do proper fit of distributions and rationale based on standard deviation</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sql_select6 =<span class="st"> </span>sql_select5
sql_from6 =<span class="st"> </span><span class="kw">str_c</span>(sql_from5, <span class="st">&quot;</span>
<span class="st">  join bioqc_gse_gpl bgl on </span>
<span class="st">  bgg.gse = bgl.gse and bg.gpl = bgl.gpl&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
sql_where6 =<span class="st"> </span><span class="kw">str_c</span>(sql_where5, <span class="st">&quot;</span>
<span class="st">  and study_median between 3 and 9&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)
res =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, <span class="kw">str_c</span>(sql_select6, sql_from6, sql_where6, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))
<span class="kw">kable</span>(res)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">GSM</th>
<th align="right">GSE</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">82933</td>
<td align="right">1837</td>
</tr>
</tbody>
</table>
<p>This is the ‘background’ of samples on which we test for tissue contamination. We store the selected samples alongside with meta information required for the analysis in the materialized view <code>BIOQC_SELECTED_SAMPLES</code> in our DBS. <a href="https://github.com/grst/BioQC_GEO_analysis/blob/master/db/views/sample_selection.sql">Here</a> is the SQL script generating the materialized view.</p>
</div>
<div id="tissue-abundance" class="section level3">
<h3><span class="header-section-number">5.3.5</span> Tissue abundance</h3>
<p>The following table shows how many samples are available for each tissue:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sqlTissue =<span class="st"> &quot;</span>
<span class="st">select /*+ parallel(16) */ tissue, count(distinct gsm) as samples from bioqc_selected_samples</span>
<span class="st">group by tissue</span>
<span class="st">order by samples desc&quot;</span>
resTissue =<span class="st"> </span><span class="kw">dbGetQuery</span>(mydb, sqlTissue)
<span class="kw">kable</span>(resTissue)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">TISSUE</th>
<th align="right">SAMPLES</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">blood</td>
<td align="right">33055</td>
</tr>
<tr class="even">
<td align="left">liver</td>
<td align="right">27029</td>
</tr>
<tr class="odd">
<td align="left">lung</td>
<td align="right">9732</td>
</tr>
<tr class="even">
<td align="left">kidney</td>
<td align="right">7588</td>
</tr>
<tr class="odd">
<td align="left">bone marrow</td>
<td align="right">6581</td>
</tr>
<tr class="even">
<td align="left">brain</td>
<td align="right">5824</td>
</tr>
<tr class="odd">
<td align="left">heart</td>
<td align="right">4819</td>
</tr>
<tr class="even">
<td align="left">breast tumor</td>
<td align="right">4238</td>
</tr>
<tr class="odd">
<td align="left">spleen</td>
<td align="right">3518</td>
</tr>
<tr class="even">
<td align="left">adipose</td>
<td align="right">2857</td>
</tr>
<tr class="odd">
<td align="left">skeletal muscle</td>
<td align="right">2474</td>
</tr>
<tr class="even">
<td align="left">skin</td>
<td align="right">2326</td>
</tr>
<tr class="odd">
<td align="left">hippocampus</td>
<td align="right">2059</td>
</tr>
<tr class="even">
<td align="left">colon</td>
<td align="right">1954</td>
</tr>
<tr class="odd">
<td align="left">hepatocyte</td>
<td align="right">1874</td>
</tr>
<tr class="even">
<td align="left">tumor</td>
<td align="right">1679</td>
</tr>
<tr class="odd">
<td align="left">white blood cells</td>
<td align="right">1551</td>
</tr>
<tr class="even">
<td align="left">lymph node</td>
<td align="right">1335</td>
</tr>
<tr class="odd">
<td align="left">breast</td>
<td align="right">1227</td>
</tr>
<tr class="even">
<td align="left">cerebellum</td>
<td align="right">1200</td>
</tr>
<tr class="odd">
<td align="left">testis</td>
<td align="right">1131</td>
</tr>
<tr class="even">
<td align="left">pbmc</td>
<td align="right">936</td>
</tr>
<tr class="odd">
<td align="left">frontal cortex</td>
<td align="right">805</td>
</tr>
<tr class="even">
<td align="left">pancreas</td>
<td align="right">770</td>
</tr>
<tr class="odd">
<td align="left">placenta</td>
<td align="right">766</td>
</tr>
<tr class="even">
<td align="left">retina</td>
<td align="right">714</td>
</tr>
<tr class="odd">
<td align="left">pancreatic islets</td>
<td align="right">647</td>
</tr>
<tr class="even">
<td align="left">thymus</td>
<td align="right">642</td>
</tr>
<tr class="odd">
<td align="left">ovary</td>
<td align="right">570</td>
</tr>
<tr class="even">
<td align="left">prostate</td>
<td align="right">523</td>
</tr>
<tr class="odd">
<td align="left">mammary gland</td>
<td align="right">479</td>
</tr>
<tr class="even">
<td align="left">hypothalamus</td>
<td align="right">470</td>
</tr>
<tr class="odd">
<td align="left">jejunum</td>
<td align="right">456</td>
</tr>
<tr class="even">
<td align="left">prefrontal cortex</td>
<td align="right">410</td>
</tr>
<tr class="odd">
<td align="left">cortex</td>
<td align="right">336</td>
</tr>
<tr class="even">
<td align="left">embryo</td>
<td align="right">290</td>
</tr>
<tr class="odd">
<td align="left">uterus</td>
<td align="right">246</td>
</tr>
<tr class="even">
<td align="left">cervix</td>
<td align="right">148</td>
</tr>
<tr class="odd">
<td align="left">stomach</td>
<td align="right">139</td>
</tr>
<tr class="even">
<td align="left">salivary gland</td>
<td align="right">133</td>
</tr>
<tr class="odd">
<td align="left">bladder</td>
<td align="right">123</td>
</tr>
<tr class="even">
<td align="left">ventral midbrain</td>
<td align="right">78</td>
</tr>
<tr class="odd">
<td align="left">eye</td>
<td align="right">69</td>
</tr>
<tr class="even">
<td align="left">adrenal gland</td>
<td align="right">67</td>
</tr>
<tr class="odd">
<td align="left">monocyte</td>
<td align="right">59</td>
</tr>
<tr class="even">
<td align="left">neuroblastoma</td>
<td align="right">45</td>
</tr>
<tr class="odd">
<td align="left">plasma</td>
<td align="right">9</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="setup-database.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="contamination-analysis.html" class="navigation navigation-next " aria-label="Next page""><i class="fa fa-angle-right"></i></a>

<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
